File: frontend/graylink-web/src/stores/modules/monitor.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import type { MonitorStatus, ApiResponse } from '@/types/api'
import { request } from '@/utils/request'

export const useMonitorStore = defineStore('monitor', () => {
  const status = ref<MonitorStatus>({
    is_running: false,
    last_check: null,
    total_files: 0,
    monitored_paths: []
  })

  const loading = ref(false)

  async function getStatus() {
    loading.value = true
    try {
      const response = await request.get<ApiResponse<MonitorStatus>>('/api/monitor/status')
      status.value = response.data.data
    } finally {
      loading.value = false
    }
  }

  async function startMonitor() {
    loading.value = true
    try {
      await request.post<ApiResponse<void>>('/api/monitor/start')
      await getStatus()
    } finally {
      loading.value = false
    }
  }

  async function stopMonitor() {
    loading.value = true
    try {
      await request.post<ApiResponse<void>>('/api/monitor/stop')
      await getStatus()
    } finally {
      loading.value = false
    }
  }

  return {
    status,
    loading,
    getStatus,
    startMonitor,
    stopMonitor
  }
})

File: backend/app/handlers/monitor.py
from fastapi import APIRouter, HTTPException
from typing import List, Optional
from pydantic import BaseModel

router = APIRouter()

class MonitorState(BaseModel):
    is_running: bool
    last_check: Optional[str] = None
    total_files: int = 0
    monitored_paths: List[str] = []

@router.get("/monitor/status")
async def get_status():
    return {
        "code": 0,
        "data": {
            "is_running": True,
            "last_check": None,
            "total_files": 0,
            "monitored_paths": []
        }
    }

@router.get("/monitor/logs")
async def get_logs(limit: int = 100):
    return {
        "code": 0,
        "data": {
            "logs": []
        }
    }

@router.post("/monitor/start")
async def start_monitor():
    return {
        "code": 0,
        "data": None
    }

@router.post("/monitor/stop")
async def stop_monitor():
    return {
        "code": 0,
        "data": None
    }

@router.post("/monitor/logs/clear")
async def clear_logs():
    return {
        "code": 0,
        "data": None
    }

@router.get("/monitor/stats")
async def get_stats():
    return {
        "code": 0,
        "data": {
            "total_checks": 0,
            "total_changes": 0,
            "last_check_duration": 0
        }
    } 